### PR TITLE
feat: add client example

### PR DESCRIPTION
**Pull request by icorderi**

### CODE REVIEW

File: examples/time_client.rs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+// Licensed under the Apache License, Version 2.0, <LICENSE-APACHE or
+// http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// http://opensource.org/licenses/MIT>, at your option. This file may not be
+// copied, modified, or distributed except according to those terms.
+
+//! A server that responds with the current time
+//!
+//! A server listening on localhost:2345. It reponds to the „now“ method, returning the current
+//! unix timestamp (number of seconds since 1.1. 1970). You can also subscribe to periodic time
+//! updates.
```
Comment (vorner) [inline examples/time_client.rs]: This doc comment probably isn't true.

File: examples/time_client.rs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    // Bind a server socket
+    let socket = TcpStream::connect(&"127.0.0.1:2345".parse().unwrap(), &handle);
+
+    let client = socket.and_then(|socket| {
+
+        // Create a client endpoint
+        let (client, _) = Endpoint::client_only(socket.framed(LineCodec::new()))
+            .logger(logger.new(o!("client" => 1)))
+            .parallel(16)
```
Comment (vorner) [inline examples/time_client.rs]: The `parallel` modifier makes no difference on client side (it affects how many requests the server is willing to process at once from a single client).

**Comment by vorner** [comment]
Thanks for the example. Can you, however, look at the failed checks (basically, rustfmt complains about some things). Also, there are some minor things in the code.

File: examples/time_client.rs (hunk 2 above)
Comment (icorderi) [inline examples/time_client.rs]: Ohh ok, removed from the example then.
Should we add a note on the docs for `parallel()`?

File: examples/time_client.rs (hunk 1 above)
Comment (icorderi) [inline examples/time_client.rs]: Copy paste bad bad bad, fixed

**Comment by icorderi** [comment]
Seems appveyor failed to download rustup :/

```sh
Build started
git clone -q https://github.com/vorner/tokio-jsonrpc.git C:\projects\tokio-jsonrpc
git fetch -q origin +refs/pull/3/merge:
git checkout -qf FETCH_HEAD
Running Install scripts
appveyor DownloadFile https://win.rustup.rs/ -FileName rustup-init.exe
Error downloading remote file: One or more errors occurred.
Inner Exception: The request was aborted: Could not create SSL/TLS secure channel.
Command exited with code 2
```

I can try to kick it out again if you want.

**Comment by vorner** [comment]
Yes, it does so from time to time. It's OK, I'll fix one or two more copy-pasted comments I just noticed and merge it.

**Comment by vorner** [comment]
OK, I merged it with a minor amend, so github wrongly things it is not merged… the technology still has a room for improvement.

**Pull request closed without merging**